test/src/test/scala/test/raphtory/DynamicClassLoaderTest.scala

```diff
@@ -85,19 +91,18 @@ class DynamicClassLoaderTest extends CatsEffectSuite {
 
   private val compiledAlgo: Generic = tb.compile(tb.parse(minimalTestCode))().asInstanceOf[Generic]
 
+  private val defaultConf: Config = ConfigBuilder.getDefaultConfig
+
+  private val waitForRaphtoryService: IO[Unit] =
+    Health.client[IO](ChannelForAddress("localhost", defaultConf.getInt("raphtory.deploy.port"))).use { client =>
+      for {
+        serving <- client.Check(HealthCheckRequest()).map(response => response.status.isServing).handleError(_ => false)
+        _       <- if (serving) IO.unit else IO.sleep(1.seconds) *> waitForRaphtoryService
+      } yield ()
+    }
+
```

Comment:
Can we implement this as part of the creation for the remote context?

Reply:
@ljeub-pometry and I just discussed this. We are going to make a proper ticket for this afterwards.